frontend/lib/resolution/Resolver.cpp

```diff
@@ -386,21 +399,93 @@ types::QualifiedType Resolver::typeErr(const uast::AstNode* ast,
   return t;
 }
 
-/* llvm::SmallVector<const Scope*> Resolver::gatherReceiverAndParentScopesForType( */
-/*     Context* context, const Type* thisType) { */
-/*   llvm::SmallVector<const Scope*> scopes; */
+/**
+  Find scopes for superclasses of a class.  The passed ID should refer to a
+  Class declaration node.  If not, this function will return an empty vector.
+
+  This function is temporary and should only be used in scopeResolveOnly mode.
+  */
+static const std::vector<const Scope*>&
```

Comment:
Unless it causes query issues this should probably also return a `ReceiverScopesVec`

Reply:
The query library already has some support for `std::vector` that writing it this way makes use of & additionally, since this is only called for classes, I'm not expecting much benefit from using `SmallVector` here. In other words, it's possible to get it working with `ReceiverScopesVec`, but I don't think it's worth the effort right now.